pkg/stringid/stringid_test.go

```diff
@@ -43,3 +43,40 @@ func TestShortenIdInvalid(t *testing.T) {
 		t.Fatalf("Id returned is incorrect: truncate on %s returned %s", id, truncID)
 	}
 }
+
+func TestAllNum(t *testing.T) {
+	tests := []struct {
+		doc, id  string
+		expected bool
+	}{
+		{
+			doc:      "mixed letters and numbers",
+			id:       "4e38e38c8ce0",
+			expected: false,
+		},
+		{
+			doc:      "letters only",
+			id:       "deadbeefcafe",
+			expected: false,
+		},
+		{
+			doc:      "numbers only",
+			id:       "012345678912",
+			expected: true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.doc, func(t *testing.T) {
+			if actual := allNum(tc.id); actual != tc.expected {
+				t.Errorf("expected %q to be %t, got %t, ", tc.id, !tc.expected, actual)
+			}
+		})
+	}
+}
+
+func BenchmarkGenerateRandomID(b *testing.B) {
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		_ = GenerateRandomID()
```

Comment:
This benchmark isn't able to measure the performance consistently as the behavior of `GenerateRandomID` is not deterministic due to usage of `crypto/rand`:

1. It has a random number of iterations (each calls `rand.Read` and performs the check)
2. The `rand.Read` itself doesn't guarantee a consistent performance.

Reply:
To make this consistent we could swap the `Reader` with a `math/rand` reader created from a source with a fixed seed (just for the benchmark).

Reply:
Ah, yeah, good point. Honestly, I initially had the separate benchmark for the "numeric" check in isolation (for this reason), and added the other benchmark mostly to see if the effect was (somewhat) measurable in the wider scope of this function;

> I did some basic benchmark to compare all of the above in isolation;
> 
> - BenchmarkParseInt: `strconv.ParseInt(TruncateID(id), 10, 64)`
> - BenchmarkParseUInt: `strconv.ParseUint(TruncateID(id), 10, 64)`
> - BenchmarkParseUIntNoTrunc: `strconv.ParseUint(id[:shortLen], 10, 64)`
> - BenchmarkAllNum: `allNum(id[:shortLen])`

Perhaps it's not worth keeping the benchmark, and I should just remove it; WDYT?

Reply:
I went ahead and dropped the first commit.

Reply:
I think we can still keep the benchmark, but perhaps it would be worth to add a comment that it's not suitable for comparing different implementations without addressing the RNG predictability first.